js/pages/ssn.js

```diff
@@ -362,19 +556,101 @@ const displayModal = (onContinue) => {
     }
 }
 
-const digitMask = (event, rawInput) => {
-
+/**
+ * Updates the raw input string based on the user's interaction with an input field.
+ * This function adjusts the raw input by adding or removing characters based on the user's input
+ * and manages the positioning of characters considering any hyphens present in the value.
+ *
+ * @param {Event} event - The input event triggered by the user's interaction with the input field.
+ * @param {string} rawInput - The current raw input string before the event's changes.
+ * 
+ * @return {string} The updated raw input string after applying the changes from the input event.
+ */
+const setRawInput = (event, rawInput) => {
+    
     const input = event.data;
-    const position = event.target.selectionStart;
+    const value = event.target.value;
+    let position = event.target.selectionStart;
+    
+    for (let i = 0; i < event.target.selectionStart; i++) {
+        if (value[i] === '-') {
+            position--;
+        }
+    }
 
     if (input !== null && /\d/.test(input)) {
         rawInput = rawInput.slice(0, position - 1) + input + rawInput.slice(position - 1);
     } else {
         rawInput = rawInput.slice(0, position) + rawInput.slice(position + 1);
     }
 
-    event.target.value = rawInput.replace(/./g, '*');
-    event.target.setSelectionRange(position, position);
-
     return rawInput;
+}
+
+/**
+ * Inserts hyphens into a string at specific positions.
+ * Hyphens are inserted after the third and fifth characters of the input string.
+ *
+ * @param {string} rawInput - The original string where hyphens are to be added.
+ * 
+ * @return {string} The modified string with hyphens inserted after the third and fifth characters.
+ */
+const addHyphens = (rawInput) => {
+
```

Comment:
```js
return rawInput.slice(0,3) + "-" + rawInput.slice(3,5) + "-" + rawInput.slice(5);
```

Reply:
While this would work if the length of the string was always a predetermined amount, this function is going to be called as the SSN is being entered. See below how when the passed in value is only 4 digits, the suggested code wouldn't output what we want:

![image](https://github.com/episphere/connectApp/assets/18222069/d3441872-e337-455d-b9ec-a035e0bd1a71)

Reply:
Maybe this?
```js
if (rawInput.length === 9) {
  return rawInput.slice(0,3) + "-" + rawInput.slice(3,5) + "-" + rawInput.slice(5);
}

return rawInput
```

Reply:
Seems unnecessary to add that code if it won't cover all cases.